### ISSUE TITLE
There is no documentation in the Android or Linux app

### ISSUE THREAD

**Issue by dgpetrie** [mhoeher/opentodolist#26]
You have invested all this time in building an app, but have provided no documentation for users to be able to take advantage of it.

There is no clues or descriptions of how the constructs of Libraries, Accounts, Lists relate.  Is there a hierarchy?  What are Libraries for collections of lists or collection of common todo items.  

When I am viewing a list and I click on schedule, how to I get back to the full list of todo items again?

I have a todo list on my linux box.  I have a webdev account setup on box.com.  Why doesn't my list show up and sync on my Android app which also has the same webdav account setup?  I have done a sync on both sides, yet there is nothing on the Android side.

You have a menu item to click on for New Library.  Where is my list of existing libraries?  What libraries are local and which ones are on one of my accounts and which account?

You are doing yourself a disservice by not giving users a high level view of how things are organized.  How do you use accounts? how do you use libraries?  To sync to I have to have a Library?  I have a list is it a Library?

Can I move a library that is local to one that is on an account?

I want to like this app as its multi-platform, but its just unusable without any user documentation on how it works and things are organized. There is not even a wiki.

**Comment by mhoeher** [comment]
Hi @dgpetrie,

thanks a lot for opening this issue and providing constructive feedback. It really helps to improve the app.

First of all: I fully agree, having good and always up-to-date documentation would be a big plus for the app.

But then, why isn't there any? Because development and maintenance of the app is basically a one man show. I am working on the app alone and hence have to decide where to put effort into. I know, this sounds like excuses only. And no: It does not mean I'm not considering documentation. Its just that development of the app is a hobby of mine - neither do users have to pay, nor are there any ads shown to indirectly earn money. So it also has to be a bit fun for me, hope this is understandable 😉

My first priority is to make the app self-explaining. For this reason, your feedback is very valuable, because I think it shows that there is lack of consistency which then leads to cases like yours, where it is unclear how things work or are supposed to work. So if the current state of the app didn't fully scare you away: Please keep on raising issues if things are unclear, really!

For now, I created this issues over on GitLab (where development of the app happens) to create and integrate documentation in the app: https://gitlab.com/rpdev/opentodolist/-/issues/379

Please don't expect this to be happening in the next one or two releases. There is at least one critical bug I am currently working on which definitely has priority over anything else. Once this is done I'll have to see when this can be scheduled.

Regarding your concrete problem with syncing: An account in OpenTodoList simply represents a connection to a backend server. However, if you already have libraries synced with the server, they won't automatically be added to the library. To add an existing library to the app, you afterwards have to use the *New Library* action from the side bar, click on the account and then then get the choice to either select an existing library to sync to your device or create a new one.

If you went through this and you still don't see anything synced to your device, this might also be an issue with the concrete WebDAV server. Unfortunately, even though WebDAV is a well-defined standard, a lot of implementations are not fully compliant here and there, so it could simply be an incompatibility between OpenTodoList and the concrete server. In this case, it'd be great of you could raise a dedicated issue to look into this.

**Comment by dgpetrie** [comment]
Thank you for the response.  I think there is a typo in your sentence below and I don't understand what you were trying to say:"However, if you already have libraries synced with the server, they won't automatically be added to the library. "
    On Sunday, November 1, 2020, 5:14:30 PM UTC, Martin Höher <notifications@github.com> wrote:  
 
 


Hi @dgpetrie,

thanks a lot for opening this issue and providing constructive feedback. It really helps to improve the app.

First of all: I fully agree, having good and always up-to-date documentation would be a big plus for the app.

But then, why isn't there any? Because development and maintenance of the app is basically a one man show. I am working on the app alone and hence have to decide where to put effort into. I know, this sounds like excuses only. And no: It does not mean I'm not considering documentation. Its just that development of the app is a hobby of mine - neither do users have to pay, nor are there any ads shown to indirectly earn money. So it also has to be a bit fun for me, hope this is understandable 😉
 
My first priority is to make the app self-explaining. For this reason, your feedback is very valuable, because I think it shows that there is lack of consistency which then leads to cases like yours, where it is unclear how things work or are supposed to work. So if the current state of the app didn't fully scare you away: Please keep on raising issues if things are unclear, really!

For now, I created this issues over on GitLab (where development of the app happens) to create and integrate documentation in the app: https://gitlab.com/rpdev/opentodolist/-/issues/379

Please don't expect this to be happening in the next one or two releases. There is at least one critical bug I am currently working on which definitely has priority over anything else. Once this is done I'll have to see when this can be scheduled.

Regarding your concrete problem with syncing: An account in OpenTodoList simply represents a connection to a backend server. However, if you already have libraries synced with the server, they won't automatically be added to the library. To add an existing library to the app, you afterwards have to use the New Library action from the side bar, click on the account and then then get the choice to either select an existing library to sync to your device or create a new one.

If you went through this and you still don't see anything synced to your device, this might also be an issue with the concrete WebDAV server. Unfortunately, even though WebDAV is a well-defined standard, a lot of implementations are not fully compliant here and there, so it could simply be an incompatibility between OpenTodoList and the concrete server. In this case, it'd be great of you could raise a dedicated issue to look into this.

**Comment by mhoeher** [comment]
Yes, you are right, that should have been "they won't automatically be added to the app", i.e. they are not automatically synced to a device after adding a new account.

IIRC this is similar to what e.g. DaAVx5 is doing: After adding an account, you still have to select the calendars to be synced with your device. Automatically adding all of them might - especially on mobiles - not be the best, because the more libraries are synced the more "energy" is consumed, i.e. batteries are drainer quicker. So it should be a concise decision by the user what to sync.

But probably the "flow" there could be improved, e.g. when adding a new account, the app could present a list of all available libraries the the user could add all or at least some of them at once.

**Comment by Bernd-Vogelgesang** [comment]
I just can second this issue over a year later:
Just found this App as a Flathub package for Linux Mint and very much liked the idea. Searched high and low for some clues how this tool is really supposed to be used.
Tried to sync it with my Android phone via Dropbox, but all fiddling around was to no avail.

Unfortunately I can only give you my evaluation: Your rational behind diverting time for development vs giving a user guidance has completely gone wrong.

The fate of so many one-man-shows ... unfortunately.

Giving up on that now.
